docs/components/index.md

```diff
@@ -1,26 +1,156 @@
+
+// Extract unique framework names
+const frameworkNames = computed(() => {
+  const names = new Set();
+  componentData.forEach(component => {
+    component.frameworks.forEach(framework => {
+      names.add(framework.name);
+    });
+  });
+  return Array.from(names);
+});
+
+// Computed property to filter components based on query and selected frameworks
+const filteredComponents = computed(() => {
+  const lowerQuery = query.value.toLowerCase();
+  const frameworks = selectedFrameworks.value;
+
+  return componentData.filter(component => {
+    const matchesQuery = component.title.toLowerCase().includes(lowerQuery);
+    const matchesFramework = frameworks.length === 0 ||
+      component.frameworks.some(framework => frameworks.includes(framework.name));
+    return matchesQuery && matchesFramework;
+  });
+});
 
+// Function to handle button clicks
+const toggleFramework = (framework) => {
+  if (selectedFrameworks.value.includes(framework)) {
+    selectedFrameworks.value = selectedFrameworks.value.filter(f => f !== framework);
+  } else {
+    selectedFrameworks.value.push(framework);
+  }
+};
+
+// Function to handle "All" button click
+const showAll = () => {
+  selectedFrameworks.value = [];
+};
 </script>
 
+<style>
+  /* remove these to custom.css and replace with warp-doc color tokens */
+  :root{
+    --w-border: #999;
+    --w-border-focus: #00F;
+    --w-button-bg: #eee;
+    --w-button-bg-active: #333;
+    --w-text: #333;
+    --w-text-inverted: #eee;
+    --w-border-radius: 4px;
+  }
+  /* set up as custom default styling somewhere else, default.css perhaps?  */
+  .input-text{
+    border: 1px solid var(--w-border);
+    color: var(--w-text);
+    padding: 4px 8px;
+    border-radius: var(--w-border-radius);
+  }
+  .input-text:focus{
+    border: 1px solid var(--w-border-focus);
+  }
+  .button{
+    background-color: var(--w-button-bg);
+    color: var(--w-text);
+    padding: 4px 8px;
+    border-radius: var(--w-border-radius);
+  }
+  .button-active{
+    background-color: var(--w-button-bg-active);
+    color: var(--w-text-inverted);
+  }
+  .button:focus{
+    outline:1px solid red;
+  }
+</style>
+
+
 # Overview
 All WARP components for Figma, React, Vue, Elements, iOS, and Android.
 
-[Filters]
-
-<cards class="grid grid-cols-1 sm:grid-cols-3">
-  <card v-for="component in componentData" :key="component.title" class="flex flex-col">
-    <h3 class="h4 text-m! static! mt-16! mx-16!">
-      <a :href="component.href" class="block before:content-empty before:absolute before:top-0 before:right-0 before:bottom-0 before:left-0 focus:outline-0">{{ component.title }}</a>
-    </h3>
-    <img class="order-first" :src="component.image.src" :alt="component.image.alt"/>
-    <p class="m-16! text-s">{{ component.description }}</p>
-  </card>
-</cards>
\ No newline at end of file
+<div>
+  <!-- Input field for text filtering -->
+  <input
+    v-model="query"
+    placeholder="Filter components"
```

Comment:
Should we use a <label> with this text instead of placeholder?

Reply:
Fixed